keps/sig-instrumentation/3498-extending-stability/README.md

```diff
@@ -0,0 +1,179 @@
+
+As an aside, much of this work has already been done, but is stashed in a local repo. 
+
+### Test Plan
+
+We have static analysis testing for stable metrics, we will extend our test coverage 
+to include metrics which are `ALPHA` and `BETA` while ignoring `INTERNAL` metrics.
+
+
+### Graduation Criteria
+
+
+#### Alpha
+
+- Kubernetes metrics framework will be enhanced to support additional stability classes
+- The static analysis pipeline of the metrics framework will be enhanced to understand how to parse more things (these are listed above)
+- All instances of `Alpha` metrics will be converted to `Internal`
+
+#### Beta
+
+- Kubernetes metrics framework will be enhanced to support marking `Alpha` and `Beta` metrics with a date. The semantics of this are yet to be determined. This date will be used to statically determine whether or not that metric should be decrepated automatically or promoted.
+- Kubernetes metrics framework will be enhanced with a script to auto-deprecate metrics which have passed their window of existence as an `Alpha` or `Beta` metric
+- It is at this point, we will determine the longevity rules for `Alpha` and `Beta` metrics
+- The `beta` stage for this framework will be a few releases. During this time, we will evaluate the utility and the ergonomics of the framework, making adjustments as necessary
+
+#### GA
+
+
+## Production Readiness Review Questionnaire
+
+During the `alpha` stage of this KEP, we will not be making any user facing changes, except marking metrics as `Internal` which were previously `Alpha`. The stability guarantees of `Internal` metrics is the same as `Alpha` currently and therefore there will not be any changes to what users can expect from the metrics they are using. 
+
+### Feature Enablement and Rollback
+
+We can revert our changes if it breaks the metrics framework. But we will be adding testing coverage as we enhance the framework, so it is unlikely that this will need to occur.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+Metrics stability framework is an internal feature of Kubernetes. 
+
+###### Does enabling the feature change any default behavior?
+
+No.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, this can be rolled back.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Metrics will be annotated with `Internal` instead of `Alpha` and vice versa.
+
+###### Are there any tests for feature enablement/disablement?
+
+No.
+
+
+### Monitoring Requirements
+
+Well, this is a meta-monitoring improvement, so it's a strange thing to monitor. But I suppose we can add metrics around how many metrics are registered divided by stability-level and metric name.
+
+
+###### How can someone using this feature know that it is working for their instance?
+
+You will see metrics from your component. 
+
```

Comment:
Please copy the rest of the PRR template here - no need to fill this in here for alpha, but I would like to see all questions here to be able to double check if something will not be needed later.

Reply:
done.